plugins/EpadDB.js

```diff
@@ -703,6 +703,138 @@ async function epaddb(fastify, options, done) {
+  //   reply.res.setHeader('Access-Control-Allow-Origin', '*');
+  //   reply.send(readStream);
+  // });
+
+  fastify.decorate('stopContainerLog', (request, reply) => {
+    const dock = new DockerService();
+    dock.stopContainerLog(`5`).then(strm => {
+      fastify.log.info('stram returned obj', strm);
+      // strm.pipe(reply.res);
+      // strm.on('data', chunk => {
+      //   strm.pipe(reply.res);
+      // });
+
+      // reply.send(strm);
+      reply.send(200);
+    });
+  });
+  fastify.decorate('getContainerLog', (request, reply) => {
+    fastify.log.info('*');
+    fastify.log.info('*');
+    fastify.log.info('*');
+    fastify.log.info('*');
+    fastify.log.info('*');
+    fastify.log.info('*'); // eslint-disable-next-line no-unused-expressions
+    // reply.send(202);
+    //  fastify.log.info('container id to call he log for : ', request);
+    fastify.getUserPluginDataPathInternal().then(pluginDataRootPath => {
+      //  fastify.log.info('pluginDataRootPath : ', pluginDataRootPath);
+      //  fastify.log.info('request', request.body);
+      const containerid = request.body.id;
+      const { creator } = request.body;
+      const dock = new DockerService();
+      // const mainPath = await dock.inspectContainer('epad_lite');
+      // fastify.log.info('main path : ', mainPath);
+      dock
+        .inspectContainer(`epadplugin_${containerid}`)
+        .then(inspectResultObject => {
+          // fastify.log.info('inspect result object', inspectResultObject);
+          // fastify.log.info('status : ', inspectResultObject.State.Status);
+          fastify.log.info(
+            `trying to read from the path : ${pluginDataRootPath}/${creator}/${containerid}/logs/logfile.txt`
+          );
+          if (inspectResultObject.State.Status === 'running') {
+            fastify.log.info('status running so sending stream');
+            reply.res.setHeader('Content-type', 'application/octet-stream');
+            reply.res.setHeader('Access-Control-Allow-Origin', '*');
+            reply.res.setHeader('connection', 'keep-alive');
+            const rdsrtm = fs.createReadStream(
+              `${pluginDataRootPath}/${creator}/${containerid}/logs/logfile.txt`
+            );
+            reply.send(rdsrtm);
+            //  rdsrtm.pipe(reply.res);
+            // dock
+            //   .getContainerLog(`${containerid}`)
+            //   .then(strm => {
+            //     fastify.log.info('stram returned obj', strm);
+            //     // strm.pipe(reply.res);
+            //     // strm.on('data', chunk => {
+            //     //   strm.pipe(reply.res);
+            //     // });
+            //     reply.send(strm);
+            //     // reply.send(strm);
+            //   })
+            //   .catch(() => {
+            //     fastify.log.info('hhhhhhyuyguyguguygugyuyguygyug');
+            //   });
+          } else {
+            reply.res.setHeader('Content-type', 'application/octet-stream');
```

Comment:
this is same with the block in catch, can be moved to a method

Reply:
rectified